applications/GeoMechanicsApplication/custom_utilities/check_utilities.cpp

```diff
@@ -25,4 +25,28 @@ void CheckUtilities::CheckDomainSize(double DomainSize, std::size_t Id, const st
         << min_domain_size << " for element " << Id << std::endl;
 }
 
+void CheckUtilities::CheckNodalVariables(const Geometry<Node>&            rGeometry,
+                                         const std::vector<VariableData>& rVariables)
+{
+    for (std::size_t i = 0; i < rGeometry.PointsNumber(); ++i) {
+		for (const auto & r_variable : rVariables){
+           if (!rGeometry[i].SolutionStepsDataHas(r_variable))
+               KRATOS_ERROR << "Missing variable " << r_variable.Name() << " on node "
+                            << rGeometry[i].Id() << std::endl;
+		}
+	}
+}
```

Comment:
If you would follow my suggestion to use `Geo::ConstVariableDataRefs`, you also need to modify the implementation a bit:
```suggestion
void CheckUtilities::CheckNodalVariables(const Geometry<Node>&            rGeometry,
                                         const Geo::ConstVariableDataRefs& rVariableRefs)
{
    for (const auto& r_node : rGeometry) {
		for (const auto & r_variable_ref : rVariableRefs){
           if (!r_node.SolutionStepsDataHas(r_variable_ref.get()))
               KRATOS_ERROR << "Missing variable " << r_variable_ref.get().Name() << " on node "
                            << r_node.Id() << std::endl;
		}
	}
}
```
Note that I have also removed the counter from the loop over the element's nodes, since a simple range-based `for` loop will do here.

Could you please apply similar changes to the implementation of `CheckNodalDof`? Thanks.

Reply:
used your suggestion with WPK idea.